src/utils.rs

```diff
@@ -67,6 +68,25 @@ pub fn extents_for_text(text: &str, family: &str, size: f64) -> Result<cairo::Te
     cr.text_extents(text).context("Couldn't create TextExtents")
 }
 
+/// Return an error if there is a collision between the keys in the window function map and the hint
+/// chars
+pub fn collision_check_window_commands<T>(
+    window_cmd_map: &HashMap<&str, T>,
+    hint_chars: &str,
+) -> Result<()> {
+    let colliding_keys: String = window_cmd_map
+        .keys()
+        .copied()
```

Comment:
Do you really need this `.copied()`? I think a reference will probably be fine here for the filter.

Reply:
I had some trouble with this, if we don't we end up with an iterator of double referenced `&&str` otherwise. `.map(|k| *k)` before the `collect()` also works?

Reply:
Eh in that case the copy is fine.